sympy/solvers/tests/test_recurr.py

```diff
@@ -184,6 +184,10 @@ def test_rsolve():
             (C0*((sqrt(-a**2 + 1) - 1)/a)**n +
              C1*((-sqrt(-a**2 + 1) - 1)/a)**n)).simplify() == 0
 
+    assert rsolve((k + 1)*y(k), y(k)) is None
+    assert (rsolve((k + 1)*y(k) + (k + 3)*y(k + 1) + (k + 5)*y(k + 2), y(k))
+            is None)
```

Comment:
> As, many RE of this form are solved RE.

what does this mean?

> In general, rsolve returns None whenever it's not able to compute a valid RE.

Yes, It didn't have a single case raising an exception other than `ValueError` , though I don't understand why is it so, One possibility is that It possibly handles a particular class of problems very well, so we don't see much failures in that domain, so no exception in that case except for some newly explored cases like these.

Your fix looks consistent with the current code, but in general silencing an exception has its own consequences, so I am not sure if it's right to silent one here.